### PR TITLE
JSON Schema Fixes Add AMENDS and required array properties

### PR DESCRIPTION
**Pull request by goneall**
Resolves issue #656

Signed-off-by: Gary O'Neall <gary@sourceauditor.com>

### CODE REVIEW

**Comment by goneall** [comment]
@tsteenbe Let me know if I should replace this with or create a second PR for development/v2.3

File: schemas/spdx-schema.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "$schema" : "http://json-schema.org/draft-07/schema#",
   "$id" : "http://spdx.org/rdf/terms",
-  "title" : "SPDX 2.2.2",
+  "title" : "SPDX 2.2",
```
Comment (seabass-labrax) [inline schemas/spdx-schema.json:4]: Did you mean to change this line?

File: schemas/spdx-schema.json [hunk 2]
```diff
@@ -709,7 +709,7 @@
...
+            "enum" : [ "VARIANT_OF", "COPY_OF", "PATCH_FOR", "TEST_DEPENDENCY_OF", "CONTAINED_BY", "DATA_FILE_OF", "OPTIONAL_COMPONENT_OF", "ANCESTOR_OF", "GENERATES", "CONTAINS", "OPTIONAL_DEPENDENCY_OF", "FILE_ADDED", "DEV_DEPENDENCY_OF", "DEPENDENCY_OF", "BUILD_DEPENDENCY_OF", "DESCRIBES", "PREREQUISITE_FOR", "HAS_PREREQUISITE", "PROVIDED_DEPENDENCY_OF", "DYNAMIC_LINK", "DESCRIBED_BY", "METAFILE_OF", "DEPENDENCY_MANIFEST_OF", "PATCH_APPLIED", "RUNTIME_DEPENDENCY_OF", "TEST_OF", "TEST_TOOL_OF", "DEPENDS_ON", "FILE_MODIFIED", "DISTRIBUTION_ARTIFACT", "AMENDS", "DOCUMENTATION_OF", "GENERATED_FROM", "STATIC_LINK", "OTHER", "BUILD_TOOL_OF", "TEST_CASE_OF", "PACKAGE_OF", "DESCENDANT_OF", "FILE_DELETED", "EXPANDED_FROM_ARCHIVE", "DEV_TOOL_OF", "EXAMPLE_OF" ]
```
Comment (seabass-labrax) [inline schemas/spdx-schema.json:712]: Looks good to me! :smiley:

File: schemas/spdx-schema.json (hunk 1 above)
Comment (goneall) [reply schemas/spdx-schema.json:4]: @seabass-labrax Yes - I'm including the minor version in the schema since it has changed since 2.2.1

**Comment by goneall** [comment]
Added a fix to this PR to include required array properties.

It adds any array property where the minimum number of elements  is greater than zero to the list of required properties for the object.

This will replace the need for PR #660

**Comment by goneall** [comment]
@seabass-labrax I added another commit - if you don't mind a re-review.

**Comment by goneall** [comment]
@dmholtz If you wouldn't mind reviewing this PR - it includes the changes from your PR #660 plus a few more missing required properties.

**Comment by goneall** [comment]
@tsteenbe @zvr If one of you could review so this so it can be merged.

File: schemas/spdx-schema.json [hunk 3]
```diff
@@ -63,7 +63,7 @@
           "type" : "string"
         }
       },
-      "required" : [ "created" ],
+      "required" : [ "created", "creator" ],
```
Comment (dmholtz) [inline schemas/spdx-schema.json:66]: I think this does not work, the property is called `creators` (plural) and that should be reflected in the list of `"required"` list

**Review by dmholtz — CHANGES_REQUESTED** [review]

File: schemas/spdx-schema.json (hunk 3 above)
Comment (goneall) [reply schemas/spdx-schema.json]: I just updated to resolve this issue

Comment (dmholtz) [reply schemas/spdx-schema.json]: 👍🏼

**Review by kestewart — APPROVED** [review]

File: schemas/spdx-schema.json [hunk 4]
```diff
@@ -553,7 +553,7 @@
             "type" : "string"
           }
         },
-        "required" : [ "SPDXID", "copyrightText", "fileName", "licenseConcluded" ],
+        "required" : [ "SPDXID", "checksums", "copyrightText", "fileName", "licenseConcluded", "licenseInfoInFiles" ],
```
[the remark this answers is not in the document]
Comment (kestewart) [reply schemas/spdx-schema.json:556]: From discussions with Gary,  this was a standard transform applied to all of them.  removing concern.

**Review by seabass-labrax — APPROVED** [review]

**Pull request merged**